src/hooks/transactions/useSignTransactions.tsx

```diff
@@ -1,189 +1,236 @@
-        dispatch(
-          moveTransactionsToSignedState({
-            sessionId,
-            status: TransactionBatchStatusesEnum.cancelled
-          })
-        );
-      }
+    /*
+    * this is triggered by abort action, 
+    * so no need to show error again
+    */
+    if (!isTxCancelled) {
+      return;
     }
+    
+    setError(errorMessage);
+
   };
 
-  async function signTransactionsWithProvider() {
+  const signTransactionsWithProvider = async () => {
+    
+    
+    const { sessionId, transactions, callbackRoute, customTransactionInformation } = transactionsToSign!;
+    const { redirectAfterSign } = customTransactionInformation;
+    const redirectRoute = callbackRoute || window.location.pathname;
+    const isCurrentRoute = window.location.pathname.includes(redirectRoute);
+    const shouldRedirectAfterSign = redirectAfterSign && !isCurrentRoute;
+
     try {
-      const {
-        sessionId,
-        transactions,
-        callbackRoute,
-        customTransactionInformation: { redirectAfterSign }
-      } = transactionsToSign!;
-      const redirectRoute = callbackRoute || window.location.pathname;
-      if (transactions?.length) {
-        const initialized = await provider.init();
-        if (!initialized) {
-          return;
-        }
-        try {
-          const signedTransactions: Transaction[] =
-            await provider.signTransactions(transactions);
-
-          const signingDisabled =
-            !signedTransactions ||
-            (signedTransactions &&
-              Object.keys(signedTransactions).length !== transactions?.length);
-          if (!signingDisabled && signedTransactions) {
-            dispatch(
-              moveTransactionsToSignedState({
-                sessionId,
-                status: TransactionBatchStatusesEnum.signed,
-                transactions: Object.values(signedTransactions).map((tx) =>
-                  parseTransactionAfterSigning(tx)
-                )
-              })
-            );
-
-            if (
-              redirectAfterSign &&
-              !window.location.pathname.includes(redirectRoute)
-            ) {
-              window.location.href = redirectRoute;
-            }
-          }
-        } catch (err) {
-          const errorMessage =
-            (error as unknown as Error)?.message ||
-            (error as string) ||
-            'error signing transaction';
-
-          console.error('error signing transaction', errorMessage);
-          onCancel(errorMessage, sessionId);
-        }
+
+      const isProviderInitialized = await provider.init();
+
+      if (!isProviderInitialized) {
+        return;
       }
-    } catch (err) {
-      const errorMessage =
-        (error as unknown as Error)?.message ||
-        (error as string) ||
-        'error signing transaction';
-      console.error('error signing transaction', errorMessage);
+
+    } catch (error) {
+
+      const customErrorMessage = 'provider not intialized';
```

Comment:
maybe we should pull all these strings inside variables outside of the component, looks a lot cleaner